version 0.6.0 (#23)
inbo/INBOtheme: vignettes/palettes.Rmd

---
title: "Available colour palettes in INBOtheme"
author: "Thierry Onkelinx"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Available colour palettes in INBOtheme}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteDepends{cluster}
---

# Introduction

This vignette demonstrates the available colour palettes in the `INBOtheme` package.
We show the palette as a plot with numbered rectangles.
The number indicates the order in the palette.

```{r knitr, include = FALSE}
library(knitr)
opts_chunk$set(
  echo = FALSE, message = FALSE, warning = FALSE
)
set.seed(20221128)
show_height <- 1.9
library(ggplot2)
```

```{r initialise, echo = TRUE}
library(INBOtheme)
```

# Palettes

## `inbo_palette()`

```{r inbo-palette, fig.height = show_height}
show_palette(inbo_palette())
```

```{r inbo-palette-demo}
demo_palette(inbo_palette())
```

## `vlaanderen_palette()`

```{r vlaanderen-palette, fig.height = show_height}
show_palette(vlaanderen_palette())
```

```{r vlaanderen-palette-demo}
demo_palette(vlaanderen_palette())
```

## `traffic_palette()`

Note that the `traffic_palette()` has a variable number of colours.

> It starts from a dark red colour and changes into a light green.

Reviewer: ```suggestion
It starts from a dark red colour and changes into a light green colour.
```

Going gradually from dark to light makes this palette readable for colour blind people.

```{r traffic-palette-9, fig.height = show_height}
show_palette(traffic_palette(9))
```

```{r traffic-palette-9-demo, fig.cap = "`traffic_palette()` with 9 levels."}
demo_palette(traffic_palette(9))
```

```{r traffic-palette-3, fig.height = show_height}
show_palette(traffic_palette(3))
```

```{r traffic-palette-3-demo, fig.cap = "`traffic_palette()` with 3 levels."}
demo_palette(traffic_palette(3))
```

\FloatBarrier

## `ordinal_palette()`

`ordinal_palette()` has a variable number of colours.
It starts with a dark colour and end with a light colour.

> Other colour are intermediate colours, resulting is a gradient from dark to light.

Reviewer: ```suggestion
Other colours are intermediate colours, resulting in a gradient from dark to light.
```

  inbo_steun_blauw, inbo_geelgr, inbo_steun_geelgroen, inbo_steun_donkerroos,
  inbo_donkergroen, inbo_oranje, inbo_geel, inbo_lichtblauw, inbo_hoofd,
  inbo_rood, inbo_grijs, inbo_bruinrood, inbo_oranje, inbo_groen
)
show_palette(inbo_named)
```

```{r inbo-named-demo}
demo_palette(inbo_named)
```

## Flanders colours

```{r vl-named, fig.height = show_height, echo = TRUE}
vl_named <- c(
  vl_black, vl_darkblue, vl_darkbrown, vl_darkgreen, vl_darkred, vl_darkyellow,
  vl_grey1, vl_grey2, vl_grey3, vl_grey4, vl_grey5, vl_grey6, vl_grey7,
  vl_lightblue, vl_lightbrown, vl_lightgreen, vl_lightred, vl_yellow
)
show_palette(vl_named)
```

```{r vl-named-demo}
demo_palette(vl_named)
```

# Simulate effect of colour blindness on the perception of palettes

## `inbo_palette()`

```{r inbo-palette-2, fig.height = show_height}
show_palette(inbo_palette())
```

```{r inbo-cluster, echo = FALSE, fig.cap = "Agglomerative hierarchical clustering of colours based on their distance."}
library(cluster)
inbo_palette() |>
  colour_blind_distance() |>
  agnes(method = "complete") |>
  plot(ask = FALSE, which.plots = 2, main = "")
```

> ```{r inbo-palette-deutan, fig.cap = "Simulation of deutan model of colorblindness."}

Reviewer 2: colorblindness is written as two words in the title but as 1 word in the figure captions. I think both can be used but bette rnot to use them both in the same text. "color blindness" has more hits on google than "colorblindness"
